src/core/api/registry.go

```diff
@@ -0,0 +1,210 @@
+package api
+
+import (
+	"fmt"
+	"net/http"
+	"strconv"
+
+	"github.com/goharbor/harbor/src/common/utils/log"
+	"github.com/goharbor/harbor/src/replication/ng/model"
+	"github.com/goharbor/harbor/src/replication/ng/registry"
+)
+
+// RegistryAPI handles requests to /api/registries/{}. It manages registries integrated to Harbor.
+type RegistryAPI struct {
+	BaseController
+	manager registry.Manager
+}
+
+// Prepare validates the user
+func (t *RegistryAPI) Prepare() {
+	t.BaseController.Prepare()
+	if !t.SecurityCtx.IsAuthenticated() {
+		t.HandleUnauthorized()
+		return
+	}
+
+	if !t.SecurityCtx.IsSysAdmin() {
+		t.HandleForbidden(t.SecurityCtx.GetUsername())
+		return
+	}
+
+	t.manager = registry.NewDefaultManager()
```

Comment:
How about just make a single instance for `Manager`? The `Prepare` will be called for every requests. And I think we will init an global `Manager` in replication model, we can use that one

Reply:
You are right.